src/Layer/OrientedImageLayer.js

```diff
@@ -0,0 +1,166 @@
+ * It is design to create an immersive view. </br>
+ * It loads a set of panoramic position and orientation,
+ * a set of camera calibration file (it's the same set of camera for each panoramic),
+ *  and a set of texture (each image for each camera for each panoramic), all organised in an {@link OrientedImageSource}. </br>
+ * It creates an {@link OrientedImageMaterial} used to do projective texture mapping on the scene.
+ * @extends GeometryLayer
+ */
+class OrientedImageLayer extends GeometryLayer {
+    /**
+     * @constructor
+     * @param { string } id - Configuration object
+     * @param { Object } config - configuration of the layer
+     * @param { OrientedImageSource } config.source - Source of the datas
+     */
+    constructor(id, config = {}) {
+        super(id, new THREE.Group(), config);
+
+        this.background = config.background || createSphere(config.sphereRadius);
+        this.background.layer = this;
+
+        this.object3d.add(this.background);
+
+        // currentPano is the current point, means it's the closest from the camera
+        this.currentPano = undefined;
+
+
+        // panos is an array of feature point, representing many panoramics.
+        // for each point, there is a position and a quaternion attribute.
+        const p1 = GeoJsonParser.parse(config.orientation, {
+            mergeFeatures: false,
+            crsOut: config.projection }).then((res) =>  {
+            this.panos = res.features;
+
+            // we need to know if we are in globe view, for orientation.
+            const isGlobe = config.projection == 'EPSG:4978';
+
+            // add position and quaternion attributes from point feature
+            for (const pano of this.panos) {
+                // set position
+                coord.set(pano.crs, pano.vertices[0], pano.vertices[1], pano.vertices[2]);
+                pano.position = coord.xyz();
+                // set quaternion
+                pano.quaternion = OrientationUtils.quaternionFromAttitude(pano.geometry[0].properties, coord, isGlobe);
+
+                // TODO clean DataSourceProvider, so that we don't have this hack to do
+                pano.material = {};
+                pano.id = pano.geometry[0].properties.id;
+            }
+        });
+
+        // array of cameras, represent the projective texture configuration for each panoramic.
+        const p2 = CameraCalibrationParser.parse(config.calibration, {}).then((c) => {
+            this.cameras = c;
+            // create the material
+            this.material = new OrientedImageMaterial(this.cameras);
+        });
+
+
+        // wait for the twos promises to tell layer is ready.
+        this.whenReady = Promise.all([p1, p2]).then(() => {
+            this.ready = true;
+            return this;
+        });
+    }
+
+    update(context) {
+        this.material.updateUniforms(context.camera.camera3D);
```

Comment:
the material should be attached to a object 3d and not to the layer

Reply:
The only goal of this layer is to create and update this material.

Reply:
Then this layer must inherit from `ColorLayer`